### PR TITLE
Use `json!` macro to simplify JSON serialization code

### PR DESCRIPTION
**Pull request by Turbo87**
see https://docs.serde.rs/serde_json/macro.json.html

I guess `+132 −349` speaks for itself 😅

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
It looks as simple as returning json in JavaScript!

File: src/controllers/keyword.rs [hunk 1]
```diff
@@ -24,22 +24,15 @@ pub fn index(req: &mut dyn RequestExt) -> EndpointResult {
...
-    let kws = data.into_iter().map(Keyword::into).collect::<Vec<_>>();
-
-    #[derive(Serialize)]
-    struct R {
-        keywords: Vec<EncodableKeyword>,
-        meta: Meta,
-    }
-    #[derive(Serialize)]
-    struct Meta {
-        total: Option<i64>,
```
[the remark this answers is not in the document]
Comment (Turbo87) [reply src/controllers/keyword.rs:36]: There was only a single codepath that filled the `Meta` struct, and that codepath always used `Some(...)`. `Some(...)` just serializes to the value within it, so it seemed a bit pointless to keep it as an `Option` if `None` is never used anywhere.

In other words: whether we use `Some(total)` or `total` results in the same response, so I preferred the simple variant.

File: src/controllers/krate/search.rs [hunk 2]
```diff
@@ -338,28 +338,16 @@ pub fn search(req: &mut dyn RequestExt) -> EndpointResult {
...
-    Ok(req.json(&R {
-        crates,
-        meta: Meta {
-            total: Some(total),
-            next_page,
-            prev_page,
+    Ok(req.json(&json!({
+        "crates": crates,
+        "meta": {
+            "total": Some(total),
```
Comment (hi-rustin) [inline src/controllers/krate/search.rs:346]: @Turbo87 So, do we need to simplify here? It also looks like it will always be Some().

Comment (Turbo87) [reply src/controllers/krate/search.rs]: good catch! :)

we don't necessarily *need* to simplify it, but I guess we should. I've rebased and fixed it in the process.

**Review by hi-rustin — APPROVED** [review]
LGTM 👍

**Comment by hi-rustin** [comment]
If there are no other objections, please feel free to r=me.

**Pull request merged**